de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/session/NegotiationHandler.java

```diff
@@ -0,0 +1,127 @@
+package de.fu_berlin.inf.dpp.server.session;
+
+import java.util.HashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+import de.fu_berlin.inf.dpp.negotiation.SessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.OutgoingSessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.IncomingSessionNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.ProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.AbstractOutgoingProjectNegotiation;
+import de.fu_berlin.inf.dpp.filesystem.IProject;
+import de.fu_berlin.inf.dpp.monitoring.NullProgressMonitor;
+import de.fu_berlin.inf.dpp.negotiation.AbstractIncomingProjectNegotiation;
+import de.fu_berlin.inf.dpp.negotiation.NegotiationTools;
+import de.fu_berlin.inf.dpp.server.progress.ConsoleProgressIndicator;
+import de.fu_berlin.inf.dpp.session.INegotiationHandler;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.util.NamedThreadFactory;
+
+public class NegotiationHandler implements INegotiationHandler {
+
+    private static final Logger log = Logger
+        .getLogger(NegotiationHandler.class);
+
+    private final ISarosSessionManager sessionManager;
+    private final ThreadPoolExecutor sessionExecutor = new ThreadPoolExecutor(0, 4, 60, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(), new NamedThreadFactory(
+        "ServerNegotiationHandler-"));
```

Comment:
You used the name twice but these are 2 different context. Session Negotiation and Resource/Project Negotiation.

Reply:
Yes, a stupid copy&paste error. fixed